src/Operation/FindAndModify.php

```diff
@@ -183,7 +180,7 @@ public function __construct($databaseName, $collectionName, array $options)
             throw InvalidArgumentException::invalidType('"writeConcern" option', $options['writeConcern'], WriteConcern::class);
         }
 
-        if (! is_bool($options['upsert'])) {
+        if (array_key_exists('upsert', $options) && ! is_bool($options['upsert'])) {
```

Comment:
The main idea of this check is to keep the same behavior of constructor verification stage. Previously there was an assumption that the array definitely contains field 'new', so the old check covered all values include NULL value.